### ISSUE TITLE
How do you train on KITTI2015?

### ISSUE THREAD

**Issue by lelelexxx** [ppliuboy/SelFlow#4]
Thanks for your excellent work!

After read your paper, I'm wondering how do you supervised train on KITTI2015 from scratch with sparse masks?  Did you downsample sparse masks or upsample optical flow in different scales to match the scale of masks? And can you sharing your train and val split on KITTI2015 train set?

Thanks again.

**Comment by ppliuboy** [comment]
For supervised training, I do not use multi-scale loss. I just upsample the final flow to the full resolution and compare it with sparse ground truth flow. Because during my experiments, I find that multi-scale losses do not make much difference. You can also add loss at different scales by downsampling sparse masks (nearest neighbor interpolation) or upsampling flow at different scales to the full resolution. 

For the train and val split on KITTI 2015 training set, I will add it in the repository.

**Comment by lelelexxx** [comment]
Thanks a lot, And I'm appreciate for your kind reply.
What data augmentation methods did you used in supervised training in KITTI2015?
I have found that, some  data augmentation methods may result Nan during supervised training on KITTI2015. And requesting again for your train and val split on KITTI2015 training set~O(∩_∩)O. 

Thanks!

**Comment by ppliuboy** [comment]
I employ very similar data augmentation strategy as previous supervised methods, such as random crop, random flip, random translation, random scale and some changes in color, brightness, contrast, gamma. Note that the ground truth on KITTI is sparse,  please be careful about some data augmentation methods, e.g., random scale, you need to use nearest neighboor sampling method for GT flow. Otherwise, you will get bad results.  Besides, you need to be careful about the direction of flow, e.g., when you employ random flip, the GT flow on the flip direction shall multiply -1.

I will release my training code with data augmentation soon.

Train and val split on KITTI 2015 are available in the [folder](https://github.com/ppliuboy/SelFlow/tree/master/img_list/KITTI).

**Comment by lelelexxx** [comment]
Thanks for your reply! I fixed the bug in my code, it is comes from the back propagation in l2_norm. Although, i'm appreciate for your kindness remainder。
Looking forward to your code~
